File: javascript/121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
 var maxProfit = function(prices) {    
    let left = 0, right = 1, maxProfit = 0
    while (right < prices.length) {
        if (prices[left] < prices[right]) {
            maxProfit = Math.max(maxProfit, prices[right] - prices[left])
        } else {
            left = right
        }
        right++
    }
    return maxProfit
};

File: golang/121-best-time-to-buy-and-sell-stock.go
package golang

func maxProfit(prices []int) int {
	left, right, maxP, n := 0, 1, 0, len(prices)
	for right < n {
		if prices[left] < prices[right] {
			profit := prices[right] - prices[left]
			if profit > maxP {
				maxP = profit
			}
		} else {
			left = right
		}
		right++
	}
	return maxP
}
